### PR TITLE
InlineRefactoring: capturing bits of legacy policy

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Rework the logic for force inline, basic block count, il size,
and maxstack so that the policy decides when these values
should inhibit inlining.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@dotnet/jit-contrib PTAL.

No diffis or inlining decision-making changes seen in mscorlib/CscBench.

**Comment by AndyAyersMS** [comment]
Looks like maxstack can be zero off-windows, but for some reason is never zero on windows. Triggers an assert. Will push a fix.

**Comment by AndyAyersMS** [comment]
Updated...

File: src/jit/inlinepolicy.cpp [hunk 1]
```diff
@@ -44,6 +44,19 @@ void LegacyPolicy::noteCandidate(InlineObservation obs)
     InlineImpact impact = inlGetImpact(obs);
     assert(impact == InlineImpact::INFORMATION);
 
+    switch (obs)
+    {
+
```
Comment (pgavlin) [inline src/jit/inlinepolicy.cpp]: Nit: looks like there's an extra newline here.

File: src/jit/inlinepolicy.cpp [hunk 2]
```diff
@@ -101,8 +115,53 @@ void LegacyPolicy::noteFatal(InlineObservation obs)
 
 void LegacyPolicy::noteInt(InlineObservation obs, int value)
 {
-    (void) value;
-    note(obs);
+    switch (obs) {
```
Comment (pgavlin) [inline src/jit/inlinepolicy.cpp]: Nit: the opening brace should be on the following line and the extra newline should be removed.

**Comment by pgavlin** [comment]
LGTM aside from a couple of style nits.

**Comment by AndyAyersMS** [comment]
OSX failed with 
```
00:19:48.309 Commencing build of mscorlib components for OSX.x64.Checked
00:19:48.695 MSBUILD : Configuration error MSB1043: The application could not start. The type initializer for 'System.IO.Path' threw an exception. Unable to find an entry point named 'GetMaximumPath' in DLL 'System.Native'.
```

File: src/jit/inlinepolicy.cpp (hunk 1 above)
Comment (AndyAyersMS) [inline src/jit/inlinepolicy.cpp]: Will fix

File: src/jit/inlinepolicy.cpp (hunk 2 above)
Comment (AndyAyersMS) [inline src/jit/inlinepolicy.cpp]: Will fix

**Comment by AndyAyersMS** [comment]
Updated

**Comment by pgavlin** [comment]
LGTM

**Pull request merged**